src/mlpack/methods/ann/rnn_impl.hpp

```diff
@@ -299,9 +299,14 @@ EvaluateWithGradient(const arma::mat& /* parameters */,
 
   double performance = 0;
   size_t responseSeq = 0;
+  size_t r_size = responses.size();
 
   for (size_t seqNum = 0; seqNum < rho; ++seqNum)
   {
+    if (seqNum >= r_size)
+    {
+      break;
```

Comment:
I think we should include a warning message here informing the user that we are skipping some steps of `bptt`

Reply:
My understanding is that when we are terminating BPTT early like this, it is effectively equivalent to if the first `rho - responses.size()` steps were all zeros.  So I don't think that a warning message is necessary there.  But I'm open to adding one if others think it is useful.  @zoq @ShikharJ any opinions on this one?